prow/slack/client.go

```diff
@@ -83,19 +83,28 @@ func (sl *Client) urlValues() *url.Values {
 	return &uv
 }
 
-func (sl *Client) postMessage(url string, uv *url.Values) ([]byte, error) {
+func (sl *Client) postMessage(url string, uv *url.Values) error {
 	resp, err := http.PostForm(url, *uv)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode != 200 {
-		t, _ := ioutil.ReadAll(resp.Body)
-		return nil, errors.New(string(t))
+	body, _ := ioutil.ReadAll(resp.Body)
+	apiResponse := struct {
+		Ok    bool   `json:"ok"`
+		Error string `json:"error"`
+	}{}
+
+	if err := json.Unmarshal(body, &apiResponse); err != nil {
+		return fmt.Errorf("API returned invalid JSON (%q): %v", string(body), err)
+	}
+
+	if resp.StatusCode != 200 || !apiResponse.Ok {
```

Comment:
Any api docs in this?

Reply:
https://api.slack.com/messaging/webhooks#handling_errors

Reply:
Though the documentation says

> in most cases you'll receive a "HTTP 200" response with a **plain text** `ok`

a successful POST results in

```
{"ok":true,"channel":"CEK044BGA","ts":"1585070782.024400","message":{"type":"message","subtype":"bot_message","text":"my test text here","ts":"1585070782.024400","username":"prow","icons":{"emoji":":prow:"},"bot_id":"B010B1B429H"}}
```

Reply:
wow, that is sad.
/lgtm
/approve

I guess you will be among the first to notice if this doesn't work anymore anyways :)